File: netlogo-headless/src/main/prim/etc/_inlinkneighbor.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.agent.{AgentSet, Turtle}
import org.nlogo.nvm.Context
import org.nlogo.nvm.Reporter

class _inlinkneighbor(private[this] val breedName: String) extends Reporter {
  def this() = this(null)

  override def toString: String = {
    return super.toString + ":" + breedName
  }

  def report(context: Context): AnyRef = {
    val parent: Turtle = context.agent.asInstanceOf[Turtle]
    val target: Turtle = argEvalTurtle(context, 0)
    val breed: AgentSet = if (breedName == null) world.links else world.getLinkBreed(breedName)
    return Boolean.box(world.linkManager.linksTo(target, parent, breed).length > 0)
  }
}

File: netlogo-gui/src/main/prim/_inlinkneighbor.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim;

import org.nlogo.agent.AgentSet;
import org.nlogo.agent.Link;
import org.nlogo.agent.Turtle;
import org.nlogo.api.LogoException;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;

public final strictfp class _inlinkneighbor
    extends Reporter {
  private final String breedName;

  public _inlinkneighbor() {
    breedName = null;
  }

  public _inlinkneighbor(String breedName) {
    this.breedName = breedName;
  }



  @Override
  public String toString() {
    return super.toString() + ":" + breedName;
  }

  @Override
  public Object report(final Context context) throws LogoException {
    Turtle target = argEvalTurtle(context, 0);
    AgentSet breed = breedName == null ? world.links() : world.getLinkBreed(breedName);
    return world.linkManager.linksTo(target, (Turtle) context.agent, breed).length > 0;
  }
}
